openbas-model/src/main/java/io/openbas/driver/MinioDriver.java

```diff
@@ -3,27 +3,41 @@
 import io.minio.BucketExistsArgs;
 import io.minio.MakeBucketArgs;
 import io.minio.MinioClient;
+import io.minio.credentials.*;
 import io.openbas.config.MinioConfig;
-import org.springframework.beans.factory.annotation.Autowired;
+import io.openbas.config.S3Config;
+import lombok.RequiredArgsConstructor;
 import org.springframework.context.annotation.Bean;
 import org.springframework.stereotype.Component;
 
 @Component
+@RequiredArgsConstructor
 public class MinioDriver {
-  private MinioConfig minioConfig;
-
-  @Autowired
-  public void setMinioConfig(MinioConfig minioConfig) {
-    this.minioConfig = minioConfig;
-  }
+  private final MinioConfig minioConfig;
+  private final S3Config s3Config;
 
   @Bean
   public MinioClient minioClient() throws Exception {
-    MinioClient minioClient =
-        MinioClient.builder()
-            .endpoint(minioConfig.getEndpoint(), minioConfig.getPort(), minioConfig.isSecure())
-            .credentials(minioConfig.getAccessKey(), minioConfig.getAccessSecret())
-            .build();
+    MinioClient minioClient;
+    if (s3Config.isUseAwsRole()) {
+      String stsEndpoint = null;
+      if (s3Config.getStsEndpoint() != null && !s3Config.getStsEndpoint().isEmpty()) {
+        stsEndpoint = s3Config.getStsEndpoint();
+      }
```

Comment:
**question:** should we throw an exception if there are null on StsEndpoint ?

Reply:
No, it's on purpose. StsEndpoint is only if you want to use a custom one (say you implement your own STS endpoint) but if you don't, the library will use either the generic global url of AWS or a regional url of AWS if your AWS_REGION variable is set.